prow/plugins/approve/approve_test.go

```diff
@@ -565,14 +599,196 @@ Approvers can cancel approval by writing `+"`/approve cancel`"+` in a comment
+			hasLabel:            false,
+			files:               []string{"a/a.go"},
+			comments:            []github.IssueComment{},
+			reviews:             []github.Review{newTestReview("Alice", "stuff", "APPROVED")},
+			selfApprove:         false,
+			needsIssue:          false,
+			lgtmActsAsApprove:   false,
+			reviewActsAsApprove: true,
+
+			expectDelete:  false,
+			expectToggle:  true,
+			expectComment: true,
+			expectedComment: `[APPROVALNOTIFIER] This PR is **APPROVED**
+
+This pull-request has been approved by: *<a href="" title="Approved">Alice</a>*
+
+The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).
+
+The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)
+
+<details >
+Needs approval from an approver in each of these files:
+
+- ~~[a/OWNERS](https://github.com/org/repo/blob/master/a/OWNERS)~~ [Alice]
+
+Approvers can indicate their approval by writing ` + "`/approve`" + ` in a comment
+Approvers can cancel approval by writing ` + "`/approve cancel`" + ` in a comment
+</details>
+<!-- META={"approvers":[]} -->`,
+		},
+		{
+			name:     "reviews in non-approving state (should not approve)",
+			hasLabel: false,
+			files:    []string{"c/c.go"},
+			comments: []github.IssueComment{},
+			reviews: []github.Review{
+				newTestReview("cjwagner", "stuff", "COMMENTED"),
+				newTestReview("cjwagner", "unsubmitted stuff", "PENDING"),
+			},
+			selfApprove:         false,
+			needsIssue:          false,
+			lgtmActsAsApprove:   false,
+			reviewActsAsApprove: true,
+
+			expectDelete:  false,
+			expectToggle:  false,
+			expectComment: true,
+			expectedComment: `[APPROVALNOTIFIER] This PR is **NOT APPROVED**
+
+This pull-request has been approved by: 
+To fully approve this pull request, please assign additional approvers.
+We suggest the following additional approver: **cjwagner**
+
+Assign the PR to them by writing ` + "`/assign @cjwagner`" + ` in a comment when ready.
+
+The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).
+
+The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)
+
+<details open>
+Needs approval from an approver in each of these files:
+
+- **[c/OWNERS](https://github.com/org/repo/blob/master/c/OWNERS)**
+
+Approvers can indicate their approval by writing ` + "`/approve`" + ` in a comment
+Approvers can cancel approval by writing ` + "`/approve cancel`" + ` in a comment
+</details>
+<!-- META={"approvers":["cjwagner"]} -->`,
+		},
+		{
+			name:     "review in request changes state means cancel",
+			hasLabel: true,
+			files:    []string{"c/c.go"},
+			comments: []github.IssueComment{
+				newTestComment("k8s-ci-robot", "[APPROVALNOTIFIER] This PR is **APPROVED**\n\nblah"),
+			},
+			reviews: []github.Review{
+				newTestReview("cjwagner", "yep", "APPROVED"),
```

Comment:
Please use the constants for `APPROVED` and `CHANGES_REQUESTED`

Reply:
Done.